lib/ansible/modules/cloud/amazon/ec2_vol.py

```diff
@@ -621,6 +621,15 @@ def main():
 
         # Add device, volume_id and volume_type parameters separately to maintain backward compatibility
         volume_info = get_volume_info(volume, state)
+
+        # deleteOnTermination is not correctly reflected on attachment
+        if module.params.get('delete_on_termination'):
+            for attempt in range(0, 8):
+                if volume_info['attachment_set'].get('deleteOnTermination') == 'true':
+                    break
+                time.sleep(5)
```

Comment:
should this use `wait` and `wait_timeout` parameters - some people might accept slightly erroneous information as a trade off for a quicker result?

Reply:
Valid criticism. I wondered about this too. time.sleep() is used all over this module though, and wait option hasn't been implemented.

Reply:
The termination deletion setting usually doesn't take all that long to propagate, so I'm ok with it as-is. I'd rather have the module take a few extra seconds than explain that wait/wait_timeout now come with data consistency implications. 

In a future fix we could make this handle wait/wait_timeout, but this is good for now.